lib/kamal/configuration/proxy.rb

```diff
@@ -30,6 +34,8 @@ def deploy_options
     {
       host: hosts,
       tls: proxy_config["ssl"].presence,
+      "tls-certificate-path": proxy_config["ssl_certificate_path"],
```

Comment:
We should mount a single volume into the proxy when we boot it.

This avoids having to set application level configuration for proxy volumes, which we want to avoid because multiple applications can share the same proxy.

We can then re-use that volume for any other features that need to provide files to the proxy (such as custom error pages for maintenance mode).

Also ideally we wouldn't need to pass paths around and kamal-proxy should be able to infer the location of the certificate.

I think it would mean that kamal-proxy would need to assume that app supplied files live in a different subdirectory to `/home/kamal-proxy/.config/kamal-proxy` so we can map the volume in separately - e.g. `$(PWD)/.kamal/proxy/apps:/home/kamal-proxy/.config/kamal-proxy-apps`.

`/home/kamal-proxy/.config/kamal-proxy` is then writable for the proxy, while `/home/kamal-proxy/.config/kamal-proxy-apps` is read only.

Then we can have `/home/kamal-proxy/.config/kamal-proxy-apps/#{service}-#{role}-#{destination}/tls/` and store the certs in there. `#{service}-#{role}-#{destination}` will match the service name passed to the proxy, so kamal-proxy can infer where the certs are stored. 

What to do think @kevinmcconnell?

Reply:
We'll also need to update `kamal app remove` to scrub the `/home/kamal-proxy/.config/kamal-proxy-apps/#{service}-#{role}-#{destination}` directory.

Reply:
The `kamal-proxy-apps` plan sounds good to me. The proxy won't really care how it's organized, as we can give it absolute paths for the files that it needs (certs, error pages, etc). But as a scheme for organizing the per-app resources I think it makes sense 👍